src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -242,11 +243,16 @@ public IEnumerable<WorkloadId> GetUpdatableWorkloadsToAdvertise(IEnumerable<Work
             }
         }
 
-        public IEnumerable<ManifestVersionUpdate> CalculateManifestRollbacks(string rollbackDefinitionFilePath)
+        public IEnumerable<ManifestVersionUpdate> CalculateManifestRollbacks(string rollbackDefinitionFilePath, WorkloadHistoryRecorder recorder = null)
         {
             var currentManifestIds = GetInstalledManifestIds();
             var manifestRollbacks = ParseRollbackDefinitionFile(rollbackDefinitionFilePath, _sdkFeatureBand);
 
+            if (recorder is not null)
+            {
+                recorder.HistoryRecord.RollbackFileContents = manifestRollbacks.ToDictionary(kvp => kvp.Id.ToString(), kvp => kvp.ManifestWithBand.Version + "/" + kvp.ManifestWithBand.Band);
```

Comment:
I'm not worried about a single duplicated line when the ManifestVersions type doesn't match, and that method isn't static, so I'd have to wrap it in a WorkloadSet (which would be confusing, since this isn't a workload set) to use it.